include/deal.II/base/parameter_handler.h

```diff
@@ -1503,6 +1503,14 @@ class ParameterHandler : public Subscriptor
   bool
   operator==(const ParameterHandler &prm2) const;
 
+  /**
+   * Asserts that all entries of the parameter handler have been found in
+   * the input file. Hence, it only makes sense to call this function after
+   * the parameters have been parsed from an input file.
+   */
+  void
+  assert_all_entries_have_been_found() const;
```

Comment:
Functions that call `Assert` are awkward to use because you can only call them in exactly one context: when you actually do want an assertion. But you have no control over what assertion that is. A better design, I think, would be if you let the function return a `bool` and then the user can decide what they want to do with the result. So maybe call it `all_entries_have_been_set()`? You'd have to update the documentation.

You might want to update the documentation to specify what "have been found" actually means. I think that that means that each parameter has been set, through one of the previous calls to any of the `parse_input()` functions (they can be called multiple times, with multiple input files) or an explicit call to `ParameterHandler::set()` (do you actually handle that as well?).

Reply:
This should be handled as well if all the code runs over the same set() function, which should be the case

Reply:
First of all I am happy that the design with the map is acceptable, I was a bit afraid that it is a bit of a hack, given that I do not have a good understanding of the ``ParameterHandler`` class. I am not sure how to resolve your first point because I want to know which parameter causes the problems so that the user immediately knows where to look at. Returning more than a bool might probably complicate code readability and documentation. In some sense I have chosen your design, because in the ``parse_input()`` function (second function of this PR) I tell the code through an additional parameter when I want to allow an assertion, and one could use a default value of ``false`` for example.

Reply:
Then return that subset of the map that corresponds to variables not yet set (e.g., returning a `std::set` that contains their keys). The user can then do with that whatever they want: if the set is empty, everything is fine. If the set is not empty, they can create an error that only contains the first unset entry (as you currently do), or outputting all unset entries.

Reply:
Would it be ok if this is done in another function so that the function with the assert can remain and the user can decide which option to call?

Reply:
I made the requested changes here https://github.com/dealii/dealii/pull/9718/commits/f7560419e84c73fa7f53ecce39a1decc7f3c67ea